corehq/messaging/smsbackends/telerivet/models.py

```diff
@@ -110,6 +114,7 @@ def send(self, msg, *args, **kwargs):
             )
 
     @classmethod
+    @quickcache(['webhook_secret'])
```

Comment:
this cache is set globally right?
If yes,
If the webhook secret is set in code then its fine but if its set on the UI, this will stay stale.
If no,
where will the cache be used?

Reply:
Good question! `TelerivetBackendForm.clean_webhook_secret()` should clear the cache.

Reply:
I'm not entirely sure I follow what the concern is?

The caching would clear after a default of 5min, according to the [docs](https://commcare-hq.readthedocs.io/caching_and_memoization.html?highlight=quickcache#quickcache), so I don't see a need to clear it manually.

Reply:
Oh, it's just good practice: It prevents confusion and support requests from users when they change a value, but HQ doesn't recognize the value they set a second ago.

Reply:
(Also, clearing the cache frees you up to change the cache timeout to longer, if that makes more sense in the context of Telerivet.)

Reply:
> Oh, it's just good practice: It prevents confusion and support requests from users when they change a value

If that's all I don't think it's too necessary to clear the cache here, since 
1) The webhook secret will realistically not change...probably ever...
2) If it does change, the function will execute again and store that value in cache
2) It does clear after 5min anyways, so no "cache clutter"
3) In production there will be several similar requests within a few seconds, each with the same BE secret, so clearing it on one request defeats the purpose of the caching...unless I am completely missing where/when the cache should be cleared?

:)

Reply:
I see.. this is going to be really tricky. If the secret is changed, tele would still have the old one for old requests and would fail anyway once the cache expires.
So seems like it should not be changed, unless compromised.

Reply:
> tele would still have the old one

When we make the request to telerivet to send the message, we give them the secret they should pass back to us on the status update request. See [here](https://github.com/dimagi/commcare-hq/blob/38dcebb9732b195b3d9657915fb20a43582f35f2/corehq/messaging/smsbackends/telerivet/models.py#L80). This is actually nice since updating the BE secret would always be up to date.

Reply:
Ya, what i meant was this
1. we send tele the secret along with SMS to be sent
2. the secret is updated on HQ
3. tele hits HQ to update status and it fails now

I am guessing the turn around would be really really quick so this should ideally never happen.

Reply:
Oh, yeah. When I tested this it hit pretty quick. I'd be impressed if you could change the secret so quick :)

Reply:
what if celery is down at tele :p